### PR TITLE
Migrate log4j2 update.

### PR DESCRIPTION
**Pull request by hasinidilanka**
**Proposed changes in this pull request**

- Remove carbon.logging and commons-logging dependency.
- Add pax-logging dependency
- Remove maven-scr plugin
- Fix linkage error in log4j2.

**When to merge this PR**
- After identity-framework is updated with log4j2 changes. 
- Need to upgrade framework version.

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/resources/log4j2.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# root loggers
+rootLogger.level = INFO
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth/src/main/resources/log4j2.properties]: Add a new line

File: components/org.wso2.carbon.identity.oidc.session/pom.xml [hunk 2]
```diff
@@ -46,9 +46,9 @@
             <groupId>commons-codec.wso2</groupId>
             <artifactId>commons-codec</artifactId>
         </dependency>
-        <dependency>
-            <groupId>org.wso2.carbon</groupId>
-            <artifactId>org.wso2.carbon.logging</artifactId>
+         <dependency>
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oidc.session/pom.xml]: Unnecessary space

File: components/org.wso2.carbon.identity.oauth/src/main/resources/log4j2.properties (hunk 1 above)
Comment (hasinidilanka) [reply components/org.wso2.carbon.identity.oauth/src/main/resources/log4j2.properties]: Fixed - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1242/commits/c3fddd3f5235a1ccf5e87591f1aa46600514119b

File: components/org.wso2.carbon.identity.oidc.session/pom.xml (hunk 2 above)
Comment (hasinidilanka) [reply components/org.wso2.carbon.identity.oidc.session/pom.xml]: Fixed - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1242/commits/c3fddd3f5235a1ccf5e87591f1aa46600514119b

**Pull request merged**